### PR TITLE
mount: add new options nocopyup|copyup for tmpfs

### PR DESCRIPTION
**Pull request by giuseppe**
add a way to disable tmpcopyup for tmpfs.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: test/e2e/run_volume_test.go [hunk 1]
```diff
@@ -116,6 +116,11 @@ var _ = Describe("Podman run with volumes", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session.ExitCode()).To(Equal(0))
 		Expect(session.OutputToString()).To(ContainSubstring("/run/test rw,nosuid,nodev,noexec,relatime - tmpfs"))
+
+		session = podmanTest.Podman([]string{"run", "--rm", "--mount", "type=tmpfs,target=/proc/sys", ALPINE, "ls", "/proc/sys"})
```
Comment (mheon) [inline test/e2e/run_volume_test.go]: Can we get new tests for tmpcopyup and notmpcopyup?

File: pkg/util/mountOpts.go [hunk 2]
```diff
@@ -80,9 +83,19 @@ func ProcessOptions(options []string, isTmpfs bool, defaults *DefaultMountOption
...
+		case "notmpcopyup":
+			if !isTmpfs {
+				return nil, errors.Wrapf(ErrBadMntOption, "the 'notmpcopyup' option is only allowed with tmpfs mounts")
+			}
+			if foundCopyUp {
+				return nil, errors.Wrapf(ErrDupeMntOption, "the 'tmpcopyup' or 'notmpcopyup' option can only be set once")
+			}
+			// do not propagate notmpcopyup to the OCI runtime
+			skipOption = true
+			foundCopyUp = true
```
Comment (rhatdan) [inline pkg/util/mountOpts.go]: Why not just continue here, rather then adding skipOption?

Comment (giuseppe) [reply pkg/util/mountOpts.go]: yes that is better, I'll update the patch

File: test/e2e/run_volume_test.go (hunk 1 above)
Comment (giuseppe) [reply test/e2e/run_volume_test.go]: yes

**Comment by giuseppe** [comment]
addressed the comments and pushed a new version :arrow_up:

File: test/e2e/run_volume_test.go [hunk 3]
```diff
@@ -116,6 +116,16 @@ var _ = Describe("Podman run with volumes", func() {
...
+
+		session = podmanTest.Podman([]string{"run", "--rm", "--mount", "type=tmpfs,target=/etc/ssl,tmpcopyup", ALPINE, "ls", "/etc/ssl"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		Expect(session.OutputToString()).To(ContainSubstring("certs"))
+
+		session = podmanTest.Podman([]string{"run", "--rm", "--mount", "type=tmpfs,target=/etc/ssl,notmpcopyup", ALPINE, "ls", "/etc/ssl"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		Expect(session.OutputToString()).To(Not(ContainSubstring("certs")))
```
Comment (TomSweeneyRedHat) [inline test/e2e/run_volume_test.go]: can you add a test to set both tmpcopyup and nottmpcopyup being set to make sure that fails?  Might be good to also add a test for each where the type is not tmpfs to ensure they fail too.

Comment (giuseppe) [reply test/e2e/run_volume_test.go]: good idea, added and pushed a new version

**Comment by giuseppe** [comment]
@TomSweeneyRedHat tests are green, new ones too

**Pull request merged**